crates/symbolicator/src/services/objects/data_cache.rs

```diff
@@ -266,3 +272,178 @@ impl CacheItemRequest for FetchFileDataRequest {
         object_handle
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::atomic::AtomicUsize;
+    use std::sync::atomic::Ordering;
+    use std::sync::Arc;
+    use std::time::Duration;
+
+    use crate::cache::Cache;
+    use crate::config::{CacheConfig, CacheConfigs, Config};
+    use crate::services::download::DownloadService;
+    use crate::services::objects::data_cache::{CacheStatus, Scope};
+    use crate::services::objects::{FindObject, ObjectPurpose, ObjectsActor};
+    use crate::sources::{FileType, HttpSourceConfig, SourceConfig, SourceId};
+    use crate::test::{self, fixture, Server};
+
+    use symbolic::common::DebugId;
+
+    use warp::filters::fs::File;
+    use warp::reject::{Reject, Rejection};
+    use warp::Filter;
+
+    #[derive(Debug, Clone, Copy)]
+    struct GoAway;
+
+    impl Reject for GoAway {}
+
+    struct FailingSymbolServer {
+        server: Server,
+        times_accessed: Arc<AtomicUsize>,
+        accept_source: SourceConfig,
+        reject_source: SourceConfig,
+        pending_source: SourceConfig,
+        not_found_source: SourceConfig,
+    }
+    impl FailingSymbolServer {
```

Comment:
On one hand, I'm always happy to see tests; On the other hand, there's enough mocking being done here that I'm wondering if this is a better fit for an integration test. Thoughts?

Reply:
I'm under the impression that blanket imports are discouraged, but I'm honestly not sure why I think that.

Reply:
Yeah, I agree that this is pretty hefty for a unit test. I'll think about moving it once we get it to actually work.